flang/runtime/io-api.cpp

```diff
@@ -1275,4 +1275,51 @@ enum Iostat IONAME(EndIoStatement)(Cookie cookie) {
   IoStatementState &io{*cookie};
   return static_cast<enum Iostat>(io.EndIoStatement());
 }
+
+template <typename INT>
+static enum Iostat CheckUnitNumberInRangeImpl(INT unit, bool handleError,
+    char *ioMsg, std::size_t ioMsgLength, const char *sourceFile,
+    int sourceLine) {
+  if (unit != static_cast<ExternalUnit>(unit)) {
```

Comment:
Do you want to `assert_static` or otherwise verify that `sizeof(INT) >= sizeof(ExternalUnit)`?

Reply:
Seems a reasonable assert. I will add it to LLVM and let it flow back here (this part is already in LLVM and is cherry-picked).

Reply:
https://reviews.llvm.org/D123285